### PR TITLE
Save/Record feature in Memory Graph

### LINKED ISSUE #337: Missing Record/Save Feature in Memory Graph
**Actual Behaviour.**  
Missing Record/Save Feature in Memory Graph.

**Expected Behaviour**  
Record/Save Feature in Memory Graph should be there.

**Additional context**  
Required by #323 

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #337 

**Changes**: Save/Record feature in Memory Graph.

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**:

### CODE REVIEW

File: app/src/main/java/io/neurolab/utilities/DataLogger.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package io.neurolab.utilities;
+
+import android.os.Environment;
+
+import java.io.File;
+
+public class DataLogger {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/utilities/DataLogger.java]: There a filepathutil you've created once. Why we creating another class for file handling? https://github.com/fossasia/neurolab-android/blob/development/app/src/main/java/io/neurolab/utilities/FilePathUtil.java

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/neurolab/utilities/DataLogger.java (hunk 1 above)
Comment (jddeep) [reply app/src/main/java/io/neurolab/utilities/DataLogger.java]: I made it actually for more manipulating functions for LoggedData files (in future) to be kept here. Okay then will move this func to FileUtils for now.

Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/utilities/DataLogger.java]: Move similar functions to similar classes.

**Comment by jddeep** [comment]
@CloudyPadmal Change made. Have a look now please.

File: app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java [hunk 2]
```diff
@@ -356,10 +368,47 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+        if (!dst.exists()) {
+            try {
+                transfer(importedFile, dst);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public static void transfer(File src, File dst) throws IOException {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java:392]: What is this doing?

Comment (jddeep) [reply app/src/main/java/io/neurolab/fragments/MemoryGraphFragment.java:392]: It actually makes a log-file/copy of the dataset being used currently with the "date-time" as name of the file in the created NeuroLab directory which is being accessed by the DataLoggerActivity. I think we can change the function name here to something more sensible and add some comments above. @CloudyPadmal what do you say?

**Review by CloudyPadmal — APPROVED** [review]

**Pull request merged**